builder/context.go

```diff
@@ -87,6 +88,11 @@ func (b *Builder) getDockerRunArgs(
 	sb.WriteString(" --volume " + volName + ":" + containerWorkspaceDir)
 	sb.WriteString(" --volume " + util.DockerSocketVolumeMapping)
 	sb.WriteString(" --volume " + homeVol + ":" + homeWorkDir)
+	if len(volMounts) > 0 {
+		for key, val := range volMounts {
+			sb.WriteString(" --volume " + key + ":" + val)
```

Comment:
lets also mount this as readonly

append `:ro` at the end of it.

https://docs.docker.com/storage/bind-mounts/#use-a-read-only-bind-mount

This will avoid any containers to update the file content.